dataverse-webapp/src/test/java/edu/harvard/iq/dataverse/dataverse/template/TemplateServiceTest.java

```diff
@@ -48,17 +49,83 @@
     @Mock
     private TemplateDao templateDao;
 
+    private int testTime = 1576156500;
+
     @BeforeEach
     public void setUp()  {
-
-        templateService.setClock(Clock.fixed(Instant.ofEpochSecond(1576156500), ZoneId.systemDefault()));
+        templateService.setClock(Clock.fixed(Instant.ofEpochSecond(testTime), ZoneId.systemDefault()));
 
         when(ejbDataverseEngine.submit(any(Command.class)))
                 .then(InvocationOnMock::getMock);
     }
 
     // -------------------- TESTS --------------------
 
+
+    @Test
+    public void createTemplate() {
+        //given
+        Dataverse dataverse = new Dataverse();
+        Template template = new Template();
+
+        //when
+        when(ejbDataverseEngine.submit(any(CreateTemplateCommand.class))).thenReturn(template);
+
+        Try<Template> createdTemplate = templateService.createTemplate(dataverse, template);
+
+        //then
+        Assert.assertTrue(createdTemplate.isSuccess());
+        Assert.assertEquals(0, createdTemplate.get().getUsageCount().longValue());
+        Assert.assertEquals(Instant.ofEpochSecond(testTime), createdTemplate.get().getCreateTime().toInstant());
+        Assert.assertTrue(dataverse.getTemplates().contains(template));
+
+    }
+
+    @Test
+    public void createTemplate_WithException() {
```

Comment:
IMO this test is not necessary. Method doesn't throw any specific exception so testing against generic failure is kinda obsolete.

Reply:
Yeah I Guess the only thing this is testing is that the exception is caught after using Command engine, but maybe it is kinda over the top.